From 74a02c6ecdf75f3dabd2ccd2d2cc92aeb32412b0 Mon Sep 17 00:00:00 2001
From: Yuta Saito <yuta@goodnotes.com>
Date: Fri, 19 Dec 2025 22:10:57 +0900
Subject: [PATCH] build: Remove `--test-wasm-stdlib` option

We already adopted wasi-libc 27, which includes the necessary fix for
null builds, so we no longer need a way to run wasm tests without
building. Also removing the logic that unintentionally enforced
`--test-wasm-stdlib` whenever `--build-wasm-stdlib` was specified.
---
 utils/build_swift/build_swift/driver_arguments.py            | 5 -----
 .../swift_build_support/build_script_invocation.py           | 4 +---
 2 files changed, 1 insertion(+), 8 deletions(-)

diff --git a/utils/build_swift/build_swift/driver_arguments.py b/utils/build_swift/build_swift/driver_arguments.py
index cf0a31a4548..f56f5afc81f 100644
--- a/utils/build_swift/build_swift/driver_arguments.py
+++ b/utils/build_swift/build_swift/driver_arguments.py
@@ -295,9 +295,6 @@ def _apply_default_arguments(args):
         args.test_xros_host = False
         args.test_android_host = False
 
-    if args.build_wasmstdlib:
-        args.test_wasmstdlib = True
-
 
 def create_argument_parser():
     """Return a configured argument parser."""
@@ -862,8 +859,6 @@ def create_argument_parser():
     option(['--build-wasm-stdlib'], toggle_true('build_wasmstdlib'),
            help='build the stdlib for WebAssembly target into a'
                 'separate build directory ')
-    option('--test-wasm-stdlib', toggle_true('test_wasmstdlib'),
-           help='test stdlib for WebAssembly')
     option(['--wasmkit'], toggle_true('build_wasmkit'),
            help='build WasmKit')
     option(['--install-wasmkit'], toggle_true('install_wasmkit'),
diff --git a/utils/swift_build_support/swift_build_support/build_script_invocation.py b/utils/swift_build_support/swift_build_support/build_script_invocation.py
index f52bf2237ef..e8ca4385407 100644
--- a/utils/swift_build_support/swift_build_support/build_script_invocation.py
+++ b/utils/swift_build_support/swift_build_support/build_script_invocation.py
@@ -685,9 +685,7 @@ class BuildScriptInvocation(object):
         builder.add_product(products.WasmKit,
                             is_enabled=self.args.build_wasmkit)
         builder.add_product(products.WasmStdlib,
-                            # Revert `or self.args.test_wasmstdlib` once we adopt `wasi-sdk-26`
-                            # or higher version that includes https://github.com/WebAssembly/wasi-libc/commit/eadb436d5c09f7983c3a687086e5af6b6e9f5510.patch
-                            is_enabled=self.args.build_wasmstdlib or self.args.test_wasmstdlib)
+                            is_enabled=self.args.build_wasmstdlib)
         builder.add_product(products.WasmThreadsStdlib,
                             is_enabled=self.args.build_wasmstdlib)
         builder.add_product(products.WasmSwiftSDK,
-- 
2.48.1

